File: 0274-h-index/0274-h-index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        non_increased_citations = sorted(citations, reverse=True)

        h_index = 0

        for index in range(len(non_increased_citations)):
            if non_increased_citations[index] >= index + 1:
                h_index = index + 1
            else:
                return h_index

        return h_index

File: 0274-h-index/0274-h-index.go
func hIndex(citations []int) int {
    sort.Ints(citations)

    for i:= len(citations)-1; i > -1; i--{
        num_paper := len(citations) - i
        if num_paper == citations[i]{
            return num_paper
        }
        if num_paper > citations[i]{
            return num_paper - 1
        }
    }

    return len(citations)
}